dali/pipeline/operator/operator.cc

```diff
@@ -16,6 +16,16 @@
 
 namespace dali {
 
+template <typename Backend>
+void OperatorBase::EnforceConstantBatchSize(const workspace_t<Backend> &ws) const {
+  DALI_ENFORCE(ws.get_input_batch_size(0) == max_batch_size_,
+               "Batch size has to be constant throughout the processing");
+}
+
+template void OperatorBase::EnforceConstantBatchSize<CPUBackend>(const workspace_t<CPUBackend>&w) const;  // NOLINT
+template void OperatorBase::EnforceConstantBatchSize<GPUBackend>(const workspace_t<GPUBackend>&w) const;  // NOLINT
+template void OperatorBase::EnforceConstantBatchSize<MixedBackend>(const workspace_t<MixedBackend>&w) const;  // NOLINT
```

Comment:
This will most likely not link with clang 6 (some DLL_PUBLIC would likely be required). Why not just keep it in the header?

Reply:
This code passes the CI, so I guess we're covered. Basically, I need an "easy" way to turn this on/off, cause I'm testing whatever is here with a script that is outside this PR. I don't want to rebuilt whole DALI, which is going to happen if that's in header file...

Reply:
The changes to TensorVector also passed CI, but didn't pass debug build in clang :(